tests/tas-mtls/src/test/java/org/alfresco/rest/MtlsRestTest.java

```diff
@@ -0,0 +1,103 @@
+package org.alfresco.rest;
+
+import javax.net.ssl.SSLHandshakeException;
+import java.io.IOException;
+import java.lang.reflect.Method;
+
+import org.alfresco.rest.core.RestWrapper;
+import org.alfresco.utility.LogFactory;
+import org.alfresco.utility.data.DataContent;
+import org.alfresco.utility.data.DataSite;
+import org.alfresco.utility.data.DataUserAIS;
+import org.alfresco.utility.model.SiteModel;
+import org.alfresco.utility.network.ServerHealth;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+@ContextConfiguration ("classpath:alfresco-mtls-context.xml")
+public abstract class MtlsRestTest extends AbstractTestNGSpringContextTests
+{
+    private static Logger LOGGER = LogFactory.getLogger();
+
+    @Autowired
+    protected ServerHealth serverHealth;
+
+    @Autowired
+    protected RestWrapper restClient;
+
+    @Autowired
+    protected DataUserAIS dataUser;
+
+    @Autowired
+    protected DataSite dataSite;
+
+    @Autowired
+    protected DataContent dataContent;
+
+    protected SiteModel testSiteModel;
+
+    CloseableHttpClient client = HttpClients.createMinimal();
+
+    @BeforeSuite (alwaysRun = true)
+    public void checkServerHealth() throws Exception
+    {
+        super.springTestContextPrepareTestInstance();
+        serverHealth.isServerReachable();
+        serverHealth.assertServerIsOnline();
+        testSiteModel = dataSite.createPublicRandomSite();
+    }
+
+    @BeforeMethod (alwaysRun=true)
+    public void showStartTestInfo(Method method)
+    {
+        LOGGER.info(String.format("*** STARTING Test: [%s] ***",method.getName()));
+    }
+
+    @AfterMethod (alwaysRun=true)
+    public void showEndTestInfo(Method method)
+    {
+        LOGGER.info(String.format("*** ENDING Test: [%s] ***", method.getName()));
+    }
+
+    @Test
+    public void checkIfMtlsIsEnabledForTransformService() throws IOException
+    {
+        CloseableHttpResponse transformServiceResponse;
+        transformServiceResponse = client.execute(new HttpGet("http://localhost:8090"));
```

Comment:
Try to get the schema (I think it should be `https`) the hostname and the port from the configuration (it should be the same source as for the `assertServerIsOnline`)

Reply:
My bad. It's not the repo url but the t-router, but still `https` should be used.